main/src/ca/uwaterloo/flix/api/lsp/provider/completion/DeltaContext.scala

```diff
@@ -17,6 +17,72 @@ package ca.uwaterloo.flix.api.lsp.provider.completion
 
 import ca.uwaterloo.flix.language.ast.Symbol
 
+/**
+  * Represents changes between ASTs.
+  */
+object DeltaContext {
+  /**
+    * Merges two DeltaContext.
+    *
+    * Removes duplicates that has an older timestamp, and deltas older than 5 minutes.
+    *
+    * Limits deltas to 1000 elements.
+    *
+    * Sorts the list with the most recent delta as first element.
+    *
+    * @param d1  the first DeltaContext.
+    * @param d2  the second DeltaContext.
+    * @return    a DeltaContext with the of changes as List[Delta],
+    *            where the most recent is the first element of the list.
+    */
+  def mergeDeltas(d1: DeltaContext, d2: DeltaContext): DeltaContext = {
+    val newDeltas = {
+      if (d2.deltas.isEmpty) {
+        d1.deltas
+      } else {
+        val (limitDeltas, _) = filterOutdated(d1.deltas ++ d2.deltas)
+            .splitAt(1000) // Limits the list to the 1000 most recent deltas
+        limitDeltas
+      }
+    }
+    DeltaContext(newDeltas)
+  }
+
+  /**
+    * Removes all duplicates, so only the most recent appear in the list.
+    *
+    * Sorts the list with the most recent delta as first element.
+    *
+    * Also removes all deltas older than 5 minutes.
+    *
+    * @param deltas the list of deltas.
+    * @return       a new list of deltas without duplicates, older deltas than 5 minutes and in sorted order.
+    */
+  private def filterOutdated(deltas: List[Delta]): List[Delta] = {
+    val deltasWithoutDuplicates = removeOlderDuplicates(deltas)
+    val fiveMinTimestamp = Differ.getCurrentTimestamp - 300000000000L
```

Comment:
What is 300000000000L? Introduce a constant and write it as a multiplication so one can see its 5 minutes, e.g. 5 * ...